apps/build-onchain-apps/src/contract/ContractSpecification.ts

```diff
@@ -0,0 +1,26 @@
+import { Abi, type Chain } from 'viem';
+import { baseGoerli } from 'viem/chains';
+import BuyMeACoffeeABI from './BuyMeACoffee';
+
+export type ContractInstance = {
+  chain: Chain;
+  address: `0x${string}`,
```

Comment:
What is this black magic

Reply:
Basically “a string that starts with `”0x”`. Compiler can only tell from string literals naturally

Reply:
Harry Potter vibes